src/library/scala/collection/IndexedSeqView.scala

```diff
@@ -42,34 +42,66 @@ object IndexedSeqView {
   @SerialVersionUID(3L)
   private final class IndexedSeqViewIterator[A](self: IndexedSeqView[A]) extends AbstractIterator[A] with Serializable {
     private[this] var current = 0
+    private[this] var remainder = self.size
     override def knownSize: Int = self.size - current
-    def hasNext = current < self.size
+    def hasNext = remainder > 0
     def next(): A =
       if (hasNext) {
         val r = self.apply(current)
         current += 1
+        remainder -= 1
         r
       } else Iterator.empty.next()
 
     override def drop(n: Int): Iterator[A] = {
-      if (n > 0) current = Math.min(self.size, current + n)
+      if (n > 0) {
+        current = Math.min(self.size, current + n)
```

Comment:
You shouldn't need to guard this against overflows. `remainder` needs to be guarded to prevent multiple chained `drop` call from overflowing it twice (back to > 0) but once `remainder == 0` the value of `current` doesn't matter anymore.